### ISSUE TITLE
CA1720 Shouldn't trigger when you're using an @ sign.

### ISSUE THREAD

**Issue by GammaFour** [dotnet/roslyn-analyzers#2262]
I've got a method signature:

```
        public async Task<IActionResult> PostAccount([FromBody] object @object)
        {
        }

```
This is throwing a CA1720.  I've provided the '@' sign which means I understand the consequences of using an object name and have chosen to use an argument that has the same name as a type.  CA1720 should be disable when the programmer uses the '@' sign.

**Comment by sharwell** [comment]
Can you explain how this proposal is different from turning off rule CA1720?

**Comment by GammaFour** [comment]
If I use **object** without the '@', _I want a warning_.  Using the '@' means I'm explicitly using something that looks like a reserved name, but isn't.  It's not in the same category as accidentally using a reserved name.

**Comment by sharwell** [comment]
@GammaFour If you use `object`, won't you get a compiler error anyway?

**Comment by GammaFour** [comment]
OK.  After reading the description, this warning isn't what I thought it was.  Sorry to waste your time.
Great product BTW.

**Comment by GammaFour** [comment]
OK. After reading the description, this warning isn't what I thought it was. Sorry to waste your time.
Great product BTW.

**Comment by sharwell** [comment]
@GammaFour No problem at all. I just wanted to make sure I understood the issue you were encountering before closing it. 👍
